integration-cli/docker_cli_run_test.go

```diff
@@ -2912,3 +2912,15 @@ func TestRunAllowPortRangeThroughPublish(t *testing.T) {
 	}
 	logDone("run - allow port range through --expose flag")
 }
+
+func TestRunOOMExitCode(t *testing.T) {
+	defer deleteAllContainers()
+
+	runCmd := exec.Command(dockerBinary, "run", "-m", "4MB", "busybox", "sh", "-c", "x=a; while true; do x=$x$x; done")
```

Comment:
This is going to be _epic_ when we have an error in our memory limit handling and this doesn't OOM. :angel: